src/FamilyHubs.ReferralApi.Infrastructure/Persistence/Repository/ApplicationDbContext.cs

```diff
@@ -35,17 +36,19 @@ IConfiguration configuration
 #endif
         _auditableEntitySaveChangesInterceptor = auditableEntitySaveChangesInterceptor;
         //DbKey should be overriden in build pipeline
-        string dbKey = configuration.GetValue<string>("DbKey");
+        string dbKey = configuration.GetValue<string>("DbKey") ?? string.Empty;
+        ArgumentNullException.ThrowIfNull(configuration);
         if (string.IsNullOrEmpty(dbKey))
         {
             throw new NotFoundException(nameof(IConfiguration), "DbKey");
         }
-
+        
         this._provider = new GenerateEncryptionProvider(dbKey);
     }
 
     protected override void OnModelCreating(ModelBuilder modelBuilder)
     {
+        modelBuilder.Entity<Referral>().Property(e => e.RequestNumber).UseIdentityAlwaysColumn();
```

Comment:
Please move this inside ReferralConfiguration class

Reply:
This is deliberate so that the database will generate the request number, so it needs to stay